pkg/api-server/inspect_endpoints.go

```diff
@@ -38,6 +38,8 @@ func (f fakeDataSourceLoader) Load(ctx context.Context, mesh string, source *sys
 	return []byte("secret"), nil
 }
 
+var CustomizeProxy func(meshContext xds_context.MeshContext, proxy *core_xds.Proxy) error
```

Comment:
There's `Runtime.Extensions()` is this maybe a less fragile and global way to implement this?

Reply:
That's a good idea to make it less fragile, but I don't think `Runtime.Extesions` is a good place for that. We discussed with @jakubdyszkiewicz previously that it would be great to use real ProxyBuilder in Inspect API, so I moved it to `Runtime` (also I had to move some interfaces to avoid circular dependencies). So now it's easy to customize ProxyBuilder if it's needed for custom policies

Reply:
Sorry, I like the original approach much more. I really don't like moving stuff to `pkg/core/xds` just to get rid of cyclic imports with runtime. The `pkg/core/xds` package ideally should be gone. I don't think `DataplaneCallbacks` belongs in it, it's specific to server, therefore it was in `pkg/xds/server/callbacks`.

I think ideally
* `pkg/core/xds` should be moved to `pkg/xds/...`
* Parts of bootstrap and runtime related to XDS should be in `pkg/xds/...`
* API endpoitns should be in `pkg/xds/rest` or `pkg/xds/api`